app/src/main/java/com/geeksville/mesh/model/UIState.kt

```diff
@@ -194,10 +195,20 @@ class UIViewModel @Inject constructor(
         nodeDB.getNodes(state.sort, state.filter, state.includeUnknown)
     }.stateIn(
         scope = viewModelScope,
-        started = WhileSubscribed(5_000),
+        started = WhileSubscribed(STOP_TIMEOUT_MILLIS),
         initialValue = emptyList(),
     )
 
+    @OptIn(ExperimentalCoroutinesApi::class)
+    fun getNodeByNum(nodeNum: Int): StateFlow<NodeInfo?> = nodeDB.nodeDBbyNum.mapLatest { nodes ->
+        nodes[nodeNum]
+    }.stateIn(
+        scope = viewModelScope,
+        started = WhileSubscribed(STOP_TIMEOUT_MILLIS),
+        initialValue = null,
+    )
```

Comment:
then this can also move to your own ViewModel

Reply:
I ended not using that function because it was overkill for what we needed it for. Instead I only retrieve the short name of the selected node. This helped avoid unnecessary re-composition.